### PR TITLE
max_pool_2d stride support

### PR DESCRIPTION
**Pull request by SinaHonari**
Issue #2196

### CODE REVIEW

**Comment by lamblin** [comment]
Can you rebase on the latest central/master? I think only one commit should be in that PR.

File: theano/tensor/signal/downsample.py [hunk 1]
```diff
@@ -235,14 +235,15 @@ def grad(self, inp, grads):
...
         maxout = self(x)
-        if self.st != self.ds:
-            return [theano.gradient.grad_not_implemented(self, 0, x)]
         return [DownsampleFactorMaxGrad(self.ds,
                                         ignore_border=self.ignore_border,
                                         st=self.st)(
                                             x, maxout, gz)]
 
     def c_code(self, node, name, inp, out, sub):
+        "No implementation is currently for the case where
```
Comment (lamblin) [inline theano/tensor/signal/downsample.py]: This should be a comment (starting with `# `), not a docstring.
Also, currently it is a parse error, as multi-line strings should be enclosed in triple marks (`'''` or `"""`).

**Comment by lamblin** [comment]
It looks like the tests you added can be quite long, and make travis time out.
I'm not sure yet what the best solution would be, let me think about it.

**Comment by nouiz** [comment]
thanks.

**Pull request merged**